app/scripts/controllers/metametrics.ts

```diff
@@ -409,11 +418,30 @@ export default class MetaMetricsController {
     const { fragments } = this.store.getState();
 
     const id = options.uniqueIdentifier ?? uuidv4();
-    const fragment = {
+    let fragment = {
       id,
       ...options,
       lastUpdated: Date.now(),
     };
+
+    /**
+     * HACK: "transaction-submitted-<id>" fragment hack
+     * A "transaction-submitted-<id>" fragment may exist following the "Transaction Added"
+     * event to persist accumulated event fragment props to the "Transaction Submitted" event
+     * which fires after a user confirms a transaction. Rejecting a confirmation does not fire the
+     * "Transaction Submitted" event. In this case, these abandoned fragments will be deleted
+     * instead of finalized with canThrowAwayIfAbandoned set to true.
+     */
+    const hasExistingSubmittedFragment =
+      options.initialEvent === TransactionMetaMetricsEvent.submitted &&
+      fragments[id];
+
+    if (hasExistingSubmittedFragment) {
+      fragment = merge(fragments[id], fragment, {
```

Comment:
Do we want to merge everything from the existing fragment, or is it safer to just copy `properties` and `sensitiveProperties` explicitly?

Reply:
I think the merge method will handle merging the properties and sensitiveProperties as we intend

Reply:
confirmed with the newly added tests